### PR TITLE
Add fields to Microsoft Sentinel mirror out

### PR DESCRIPTION
**Pull request by aaron1535**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-10061)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/AzureSentinel/Layouts/layoutscontainer-Microsoft_Sentinel.json [hunk 1]
```diff
@@ -314,21 +327,51 @@
...
+						"items": [
+							{
+								"args": {},
+								"buttonClass": "primary",
+								"endCol": 4,
+								"fieldId": "",
+								"height": 44,
+								"id": "92aae170-5331-11ef-90d3-736eac4b13d9",
+								"index": 0,
+								"name": "Add New Comment",
```
Comment (JudahSchwartz) [inline Packs/AzureSentinel/Layouts/layoutscontainer-Microsoft_Sentinel.json:343]:
```suggestion
								"name": "Add Comment",
```

File: Packs/AzureSentinel/ReleaseNotes/1_6_0.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+
+#### Scripts
+
+##### New: MicrosoftSentinelSetOwnerButton
+
+- New: This script is used to set owner in remote system.
```
Comment (JudahSchwartz) [inline Packs/AzureSentinel/ReleaseNotes/1_6_0.md:6]:
```suggestion
- New: This script is used by the *Microsoft Sentinel - default layout* to set the owner in Microsoft Sentinel.
```

File: Packs/AzureSentinel/ReleaseNotes/1_6_0.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+#### Incident Fields
+
+- **Microsoft Sentinel Owner** - The owner field locked to changes. to change the owner field u can use the 'Set Remote Owner' button.
+
+#### Layouts
+
+##### Microsoft Sentinel - default layout
+
+- Improved layout. Now you can edit the owner field and add new comments.
```
Comment (JudahSchwartz) [inline Packs/AzureSentinel/ReleaseNotes/1_6_0.md:24]:
```suggestion
- You can now set the owner and add comments.
```

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelConvertCommentsToTable/MicrosoftSentinelConvertCommentsToTable.py [hunk 4]
```diff
@@ -1,17 +1,16 @@
 import demistomock as demisto
```
Comment (JudahSchwartz) [inline Packs/AzureSentinel/Scripts/MicrosoftSentinelConvertCommentsToTable/MicrosoftSentinelConvertCommentsToTable.py:1]: why do we need any changes in this file at this point?

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelConvertCommentsToTable/MicrosoftSentinelConvertCommentsToTable.py [hunk 5]
```diff
@@ -26,13 +25,13 @@ def convert_to_table(context_results: str) -> CommandResults:
...
     context_formatted = [
-        format_comment(comment) for comment in context_results
+        format_comment(comment) for comment in context_results  # type: ignore
     ]
 
     md = tableToMarkdown(
         '',
         context_formatted,
-        headers=[*dict.fromkeys(chain.from_iterable(context_formatted))],
+        headers=["message", "createdTime", "name"],
```
Comment (JudahSchwartz) [inline Packs/AzureSentinel/Scripts/MicrosoftSentinelConvertCommentsToTable/MicrosoftSentinelConvertCommentsToTable.py:34]: why this change?

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
+args:
+- name: owner_email
+  required: true
+  description: The owner email to set as user principal name.
+commonfields:
+  id: MicrosoftSentinelSetOwnerButton
```
Comment (JudahSchwartz) [inline Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.yml:6]:
```suggestion
  id: MicrosoftSentinelSetOwner
```
and change names of files not to be specific to button. THis script can also be called not by the button

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+    Returns:
+        CommandResults: Includes a markdown-formatted string with the updated owner details,
+                         and the updated owner information.
+    """
+    incident_id = context_results.get("CustomFields").get("sourceid")  # type: ignore
+    instance_name = context_results.get("sourceInstance")
+    user_principal_email = demisto.args().get("owner_email")
+
+    demisto.debug(f"set owner remote incident with owner email {user_principal_email}")
```
Comment (JudahSchwartz) [inline Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.py:21]: log all the other arguments here too

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    readable_output = tableToMarkdown(
+        f"Updated incident {incident_id} with new owner",
+        remote_owner,
+        headers=list(remote_owner.keys()),
+        headerTransform=pascalToSpace,
+        removeNull=True,
+    )
+    return CommandResults(
+        readable_output=readable_output,
+        outputs_prefix="AzureSentinel.Incident.Owner",
```
Comment (JudahSchwartz) [inline Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.py:40]: cant we just pass the exact context we got from executeCommand? We should do this instead

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.py [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+        outputs=remote_owner,
+        raw_response=result,
+    )
+
+
+def main():  # pragma: no cover
+    context = dict_safe_get(demisto.callingContext, ["context", "Incidents", 0], {})
+
+    if not context:
+        return_error("No data to present")
```
Comment (JudahSchwartz) [inline Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.py:50]: what happens if this script is called manually from the war room? Is this info still passed? Lets provide an optional argument for remote instance id

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.yml [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+  version: -1
+comment: This script is used to set owner in remote system.
+name: MicrosoftSentinelSetOwnerButton
+outputs:
+  - contextPath: AzureSentinel.Incident.Owner.assignedTo
```
Comment (JudahSchwartz) [inline Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.yml:11]: adjust output according to requested change above

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.yml [hunk 11]
```diff
@@ -0,0 +1,30 @@
+args:
+- name: owner_email
+  required: true
+  description: The owner email to set as user principal name.
```
Comment (JudahSchwartz) [inline Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.yml:4]: explain how this can be found in the sentinel ui

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/README.md [hunk 12]
```diff
@@ -0,0 +1,30 @@
+This script is used to set owner in remote system.
```
Comment (JudahSchwartz) [inline Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/README.md:1]:
```suggestion
This script can be run from the warroom or used by a layout to set the owner field in Microsoft Sentinel.
```

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/README.md [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+| Tags | dynamic-section |
+| Cortex XSOAR Version | 5.5.0 |
+
+## Inputs
+
+---
+
+| **Argument Name** | **Description** |
+| --- | --- |
+| owner_email | The owner email to set as user principal name. |
```
Comment (JudahSchwartz) [inline Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/README.md:19]: is email the same as user principal name?

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSubmitNewComment/MicrosoftSentinelSubmitNewComment.yml [hunk 14]
```diff
@@ -0,0 +1,27 @@
+args:
+- name: new_comment
+  required: true
+  description: The comment text to be added to the incident comments.
+commonfields:
+  id: MicrosoftSentinelSubmitNewComment
+  version: -1
+comment: This script is used to add a new comment to incident.
```
Comment (JudahSchwartz) [inline Packs/AzureSentinel/Scripts/MicrosoftSentinelSubmitNewComment/MicrosoftSentinelSubmitNewComment.yml:8]:
```suggestion
comment: This script is used to add a new comment to the incident.
```

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSubmitNewComment/MicrosoftSentinelSubmitNewComment.yml [hunk 15]
```diff
@@ -0,0 +1,27 @@
+args:
+- name: new_comment
```
Comment (JudahSchwartz) [inline Packs/AzureSentinel/Scripts/MicrosoftSentinelSubmitNewComment/MicrosoftSentinelSubmitNewComment.yml:2]: lets also add an optional argument here for incident id. When not provided get from the calling context

**Review by JudahSchwartz — COMMENTED** [review]
amazing!! a couple comments here. lets address these and then send to the customer for testing and tag doc review

File: Packs/AzureSentinel/Layouts/layoutscontainer-Microsoft_Sentinel.json (hunk 1 above)
Comment (aaron1535) [reply Packs/AzureSentinel/Layouts/layoutscontainer-Microsoft_Sentinel.json:343]: Done

File: Packs/AzureSentinel/ReleaseNotes/1_6_0.md (hunk 2 above)
Comment (aaron1535) [reply Packs/AzureSentinel/ReleaseNotes/1_6_0.md:6]: Done

File: Packs/AzureSentinel/ReleaseNotes/1_6_0.md (hunk 3 above)
Comment (aaron1535) [reply Packs/AzureSentinel/ReleaseNotes/1_6_0.md:24]: Done

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.yml (hunk 6 above)
Comment (aaron1535) [reply Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.yml:6]: Done

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelConvertCommentsToTable/MicrosoftSentinelConvertCommentsToTable.py (hunk 4 above)
Comment (aaron1535) [reply Packs/AzureSentinel/Scripts/MicrosoftSentinelConvertCommentsToTable/MicrosoftSentinelConvertCommentsToTable.py:1]: The current script usually returns an error
and returns the irrelevant information

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.py (hunk 8 above)
Comment (aaron1535) [reply Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.py:40]: possible
I thought I would present things in a nicer and cleaner way

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.py (hunk 9 above)
Comment (aaron1535) [reply Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.py:50]: True for both instance_name and incident_id

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.yml (hunk 10 above)
Comment (aaron1535) [reply Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.yml:11]: Simply put here the outputs that appear in the executed command?

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.py (hunk 9 above)
Comment (aaron1535) [reply Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.py:50]: Done

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/README.md (hunk 13 above)
Comment (aaron1535) [reply Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/README.md:19]: Yes 
its the same one

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSubmitNewComment/README.md [hunk 16]
```diff
@@ -0,0 +1,29 @@
+This script is used to add a new comment to incident.
```
[the remark this answers is not in the document]
Comment (aaron1535) [reply Packs/AzureSentinel/Scripts/MicrosoftSentinelSubmitNewComment/README.md:1]: Done

[the remark this answers is not in the document]
Comment (aaron1535) [reply Packs/AzureSentinel/Scripts/MicrosoftSentinelSubmitNewComment/README.md:1]: is this a question?

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSubmitNewComment/MicrosoftSentinelSubmitNewComment.yml (hunk 15 above)
Comment (aaron1535) [reply Packs/AzureSentinel/Scripts/MicrosoftSentinelSubmitNewComment/MicrosoftSentinelSubmitNewComment.yml:2]: Done

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSubmitNewComment/MicrosoftSentinelSubmitNewComment.yml (hunk 14 above)
Comment (aaron1535) [reply Packs/AzureSentinel/Scripts/MicrosoftSentinelSubmitNewComment/MicrosoftSentinelSubmitNewComment.yml:11]: Done

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/README.md (hunk 12 above)
Comment (aaron1535) [reply Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/README.md]: Done

File: Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.yml (hunk 11 above)
Comment (aaron1535) [reply Packs/AzureSentinel/Scripts/MicrosoftSentinelSetOwnerButton/MicrosoftSentinelSetOwnerButton.yml]: Done
